### PR TITLE
Switch to GitHub Actions for CI

### PR DESCRIPTION
**Pull request by bsideup**
This includes removing `.travis.yml` file

### CODE REVIEW

**Review by ericbottard — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: CI
+
+on:
+    pull_request: {}
+    push: { branches: [ master, "1.2.x" ] }
```
Comment (simonbasle) [inline .github/workflows/ci.yml:5]: is the second one a regex pattern? don't we want to also build `1.1.x` branch, and in the future build `1.3.x` branch without having to touch this file ?

**Review by simonbasle — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bsideup) [reply .github/workflows/ci.yml:5]: no, it is a concrete value ("1.2.x"). GHA support patterns, I can change to to a pattern if you wish :)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: CI
+
+on:
+    pull_request: {}
+    push:
+      branches:
+        - master
+        - 1.*
```
Comment (simonbasle) [inline .github/workflows/ci.yml:8]:
```suggestion
        - *.*.x
```

Comment (ericbottard) [reply .github/workflows/ci.yml:8]: Beware if you start the pattern with a star: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

Comment (bsideup) [reply .github/workflows/ci.yml:8]: I would not do that, yeah. 2.0 is going to be a big change anyways, so it is okay to add a line to the CI file by the time it is happening :)

Comment (simonbasle) [reply .github/workflows/ci.yml:8]: how about this safer pattern:
```suggestion
        - [0-9]+.[0-9]+.x
```

Comment (simonbasle) [reply .github/workflows/ci.yml:8]: Last suggestion (what I'm doing in core's latest PR, turns out the quotes are necessary here too):
```suggestion
        - "[0-9].[0-9]+.x"
```

**Review by simonbasle — APPROVED** [review]

**Review by simonbasle — CHANGES_REQUESTED** [review]
there's a travis.yml file that needs to be removed.

**Review by simonbasle — APPROVED** [review]

**Pull request merged**